src/components/KTable/KTable.vue

```diff
@@ -515,44 +515,60 @@ export default defineComponent({
      * Grabs listeners from attrs matching a prefix to attach the
      * event that is dynamic. e.g. `v-on:cell:click`, `@row:focus` etc.
      * @param {String} prefix - event listener prefix e.g. `row:`, `cell:`
-     * @param {any} listeners - attrs on the vue instance to pluck from
+     * @param {any} attrs - attrs on the vue instance to pluck from
      * @returns {Function} - returns a function that can pass an entity to the
                              listener callback function.
     */
-    const pluckListeners = (prefix: any, listeners: any): any => {
-      return (entity: any, type: any) =>
-        Object.keys(listeners).reduce((acc: any, curr) => {
+    const pluckListeners = (prefix: any, attrs: any): any => {
+      return (entity: any, type: any) => {
+        const onRE = /^on[^a-z]/
+        const listeners = {} as any
+
+        for (const property in attrs) {
+          if (onRE.test(property)) {
+            listeners[property] = attrs[property]
+          }
+        }
+
+        return Object.keys(listeners).reduce((acc: any, curr) => {
           if (curr.indexOf(prefix) === 0) {
             const parts = curr.split(prefix)
             acc[parts[1]] = (e: any) => listeners[curr](e, entity, type)
           }
           return acc
         }, {})
+      }
     }
 
-    const tdlisteners = computed(() => { // TODO: can we replace with something else??
-      return pluckListeners('cell:', attrs)
-    })
+    const tdlisteners = computed(() => {
+      return (entity: any, rowData: any) => {
+        const rowListeners = pluckListeners('onRow:', attrs)(rowData, 'row')
+        const cellListeners = pluckListeners('onCell:', attrs)(entity, 'cell')
 
-    const trlisteners = computed(() => {
-      return (entity: any, type: any) => {
-        const pluckedListeners = pluckListeners('row:', attrs)(entity, type)
-        if (pluckedListeners.click) {
+        if (rowListeners.click) {
           isClickable.value = true
         }
+
         return {
-          ...pluckedListeners,
+          ...rowListeners,
+          ...cellListeners,
           click(e: any) {
-            if (e.target.tagName === 'TD' && pluckedListeners.click) {
-              pluckedListeners.click(e, entity, type)
+            if (e.target.tagName === 'TD' && (rowListeners.click || cellListeners.click)) {
```

Comment:
If this doesn't work after merge, we can try removing this line (as @dustinryerson suggested)

`e.target.tagName === 'TD'`